src/submodule.c

```diff
@@ -327,7 +327,7 @@ static int submodules_from_index(git_strmap *map, git_index *idx, git_config *cf
        const git_index_entry *entry;
        git_buf name = GIT_BUF_INIT;
 
-       if ((error = git_iterator_for_index(&i, idx, NULL)) < 0)
+       if ((error = git_iterator_for_index(&i, git_index_owner(idx), idx, NULL)) < 0)
```

Comment:
`git_index_owner(idx)` will never return `NULL` here, as the only place `idx` can be coming from is through `git_repository_index(&idx, repo)`.